src/bun.js/node-tls.exports.js

```diff
@@ -78,31 +117,110 @@ const TLSSocket = (function (InternalTLSSocket) {
       throw Error("Not implented in Bun yet");
     }
 
-    emit(event, args) {
-      super.emit(event, args);
-
-      if (event === "connect" && !this._readableState?.destroyed) {
-        this.authorized = true;
-        this.secureConnecting = false;
-        this._secureEstablished = true;
-        this._securePending = false;
-
-        super.emit("secureConnect", args);
-      }
-    }
-
     [buntls](port, host) {
       var { servername } = this;
       if (servername) {
         return {
           serverName: typeof servername === "string" ? servername : host,
+          ...this.#secureContext,
         };
       }
 
       return true;
     }
   },
 );
+
+class Server extends NetServer {
+  key;
+  cert;
+  ca;
+  passphrase;
+  secureOptions;
+  _rejectUnauthorized;
```

Comment:
Does Node expose `_requestCert` and `_rejectUnauthorized`? If not can we make this `#rejectUnauthorized` and `#requestCert`